src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesVsPatterns.cs

```diff
@@ -0,0 +1,1336 @@
+    {
+        var t = (o, o);
+        _ = t switch
+        {
+            (1, 2) => 1,
+            ({}, {}) => 2,
+        };
+    }
+    void M3(object o)
+    {
+        var t = (o, o);
+        _ = t switch
+        {
+            (null, 2) => 1,
+            ({}, {}) => 2,
+            (null, {}) => 3,
+        };
+    }
+    void M4(object o)
+    {
+        var t = (o, o);
+        _ = t switch // 2 not exhaustive
+        {
+            { Item1: null, Item2: 2 } => 1,
+            { Item1: {}, Item2: {} } => 2,
+        };
+    }
+    void M5(object o)
+    {
+        var t = (o, o);
+        _ = t switch
+        {
+            { Item1: 1, Item2: 2 } => 1,
+            { Item1: {}, Item2: {} } => 2,
+        };
+    }
+    void M6(object o)
+    {
+        var t = (o, o);
+        _ = t switch
+        {
+            { Item1: null, Item2: 2 } => 1,
+            { Item1: {}, Item2: {} } => 2,
+            { Item1: null, Item2: {} } => 3,
+        };
+    }
+    void M7(object o, bool b)
+    {
+        _ = o switch // 3 not exhaustive
+        {
+            null when b => 1,
+            {} => 2,
+        };
+    }
+    void M8(object o, bool b)
+    {
+        _ = o switch
+        {
+            null when b => 1,
+            {} => 2,
+            null => 3,
+        };
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (8,15): warning CS8655: The switch expression does not handle some null inputs (it is not exhaustive).
+                //         _ = t switch // 1 not exhaustive
+                Diagnostic(ErrorCode.WRN_SwitchExpressionNotExhaustiveForNull, "switch").WithLocation(8, 15),
+                // (36,15): warning CS8655: The switch expression does not handle some null inputs (it is not exhaustive).
+                //         _ = t switch // 2 not exhaustive
+                Diagnostic(ErrorCode.WRN_SwitchExpressionNotExhaustiveForNull, "switch").WithLocation(36, 15),
+                // (63,15): warning CS8655: The switch expression does not handle some null inputs (it is not exhaustive).
+                //         _ = o switch // 3 not exhaustive
+                Diagnostic(ErrorCode.WRN_SwitchExpressionNotExhaustiveForNull, "switch").WithLocation(63, 15));
+        }
+
+        [Fact, WorkItem(31881, "https://github.com/dotnet/roslyn/issues/31881")]
+        public void NullableVsPattern_31881()
+        {
+            var source = @"
+using System;
+#nullable enable
+
+public class C
+{
+    public object? AProperty { get; set; }
+    public void M(C? input, int i)
+    {
+        if (input?.AProperty is string str)
+        {
+            Console.WriteLine(str.ToString());
+
+            switch (i)
+            {
+                case 1:
+                    Console.WriteLine(input?.AProperty.ToString());
+                    break;
+                case 2:
+                    Console.WriteLine(input.AProperty.ToString());
```

Comment:
By our rules, I would have expected this to produce a warning since the previous case did a conditional deference. Am I misremembering our rules?

Reply:
There is no control-flow path from there to here.

---
In reply to: [267961099](https://github.com/dotnet/roslyn/pull/34273#discussion_r267961099) [](ancestors = 267961099)

Reply:
That matches my memory as well. When there is a `null` check in the `case` directly then it affects the state of the condition. But conditional access in a case block doesn't flow back up. 

---
In reply to: [268373938](https://github.com/dotnet/roslyn/pull/34273#discussion_r268373938) [](ancestors = 268373938,267961099)